jupyter_server/services/contents/filemanager.py

```diff
@@ -82,14 +82,19 @@ def _validate_post_save_hook(self, proposal):
             value = import_item(value)
         if not callable(value):
             raise TraitError("post_save_hook must be callable")
+        self._post_save_hooks.append(value)
         return value
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._post_save_hooks = []
+
     def run_post_save_hook(self, model, os_path):
         """Run the post-save hook if defined, and log errors"""
-        if self.post_save_hook:
+        for post_save_hook in self._post_save_hooks:
             try:
                 self.log.debug("Running post-save hook on %s", os_path)
-                self.post_save_hook(os_path=os_path, model=model, contents_manager=self)
+                post_save_hook(os_path=os_path, model=model, contents_manager=self)
             except Exception as e:
                 self.log.error("Post-save hook failed o-n %s", os_path, exc_info=True)
```

Comment:
would it be helpful to log the name of the `hook` which failed.?

Reply:
Maybe, although it would be displayed as something like `<function foo at 0x7fd9f2a2f2e0>`.

Reply:
see if you can use `__name__`  syntax
```
In [4]: def test123():
   ...:     print("hi")
   ...: 

In [5]: func = test123

In [6]: func
Out[6]: <function __main__.test123()>

In [7]: print(func)
<function test123 at 0x7fdaad0d2790>

In [8]: print(func.__name__)
test123
```